File: Python/dinheiro.py
valor = float(input("Valor: "))
pago = float(input("Pago: "))

troco = round((pago - valor) * 100)

m25 = m10 = m5 = m1 = 0

if troco > 0:
    m25 = troco // 25
    troco %= 25

    m10 = troco // 10
    troco %= 10

    m5 = troco // 5
    troco %= 5

    m1 = troco

print(f"Moedas de 25: {int(m25)}")
print(f"Moedas de 10: {int(m10)}")
print(f"Moedas de 5: {int(m5)}")
print(f"Moedas de 1: {int(m1)}")

File: Linguagem C/dinheiro.c
#include <stdio.h>
#include <math.h> 

int main(void)
{
    float valor = 0;
    float pago = 0;
    int m25 = 0;
    int m10 = 0;
    int m5 = 0;
    int m1 = 0;

    printf("\nValor: ");
    scanf("%f", &valor);

    printf("\nPago: ");
    scanf("%f", &pago);

    int troco = (int) round((pago - valor) * 100);

    if (troco > 0)
    {
        m25 = troco / 25;
        troco %= 25;

        m10 = troco / 10;
        troco %= 10;

        m5 = troco / 5;
        troco %= 5;

        m1 = troco;
    }

    printf("\nMoedas de 25: %i", m25);
    printf("\nMoedas de 10: %i", m10);
    printf("\nMoedas de 5: %i", m5);
    printf("\nMoedas de 1: %i", m1);

    return 0;
}
